### PR TITLE
librocksdb-sys: Add support for building with io_uring on Linux.

### PR DESCRIPTION
**Pull request by parazyd**

### CODE REVIEW

File: librocksdb-sys/build.rs [hunk 1]
```diff
@@ -198,6 +198,11 @@ fn build_rocksdb() {
         config.define("WITH_JEMALLOC", "ON");
     }
 
+    if target.contains("linux") && cfg!(feature = "io-uring") {
+        pkg_config::probe_library("liburing").unwrap();
```
Comment (aleksuss) [inline librocksdb-sys/build.rs:202]: What will be if there is no `liburing` library on a host?

```suggestion
        if pkg_config::probe_library("liburing").is_ok() {
            config.define("ROCKSDB_IOURING_PRESENT", Some("1"));
        }
```

Comment (aleksuss) [inline librocksdb-sys/build.rs:202]: What will be if there is no `liburing` library on a host?

```suggestion
        if pkg_config::probe_library("liburing").is_ok() {
            config.define("ROCKSDB_IOURING_PRESENT", Some("1"));
        }
```

**Comment by aleksuss** [comment]
It would be good to add a new job in a CI for checking this feature.

**Comment by parazyd** [comment]
@aleksuss I thought it's better not to silently fall through if the feature is explicitly enabled but the library is unavailable. Would you like me to chane this?

**Comment by aleksuss** [comment]
Then you probably should add a message to inform a user: 
```rust
pkg_config::probe_library("liburing").except("The io_uring feature was provided but the library was not found");
```
or something...

**Comment by parazyd** [comment]
Updated. Thanks.

File: librocksdb-sys/build.rs [hunk 2]
```diff
@@ -218,6 +218,12 @@ fn build_rocksdb() {
         config.define("WITH_JEMALLOC", "ON");
     }
 
+    if target.contains("linux") && cfg!(feature = "io-uring") {
```
Comment (aleksuss) [inline librocksdb-sys/build.rs:221]:
```suggestion
    #[cfg(feature = "io-uring")]
    if target.contains("linux") && cfg!(feature = "io-uring") {
```

File: librocksdb-sys/build.rs [hunk 3]
```diff
@@ -218,6 +218,12 @@ fn build_rocksdb() {
...
+        pkg_config::probe_library("liburing")
+            .except("The io-uring feature was requested but the library is not available");
```
Comment (aleksuss) [inline librocksdb-sys/build.rs:223]:
```suggestion
            .expect("The io-uring feature was requested but the library is not available");
```

**Review by aleksuss — CHANGES_REQUESTED** [review]
Fix the typo in method `expect` please.

**Review by stanislav-tkach — APPROVED** [review]

**Review by aleksuss — APPROVED** [review]

**Pull request merged**